File: task_equivalents/llm_translations/gemini/optimized/python/statistical_modelling_analysis/02_logistic_regression_odds_ratios.py
import pandas as pd
import statsmodels.api as sm

Weekly = pd.read_csv('data/csv/Weekly.csv', index_col=0)

# R's 1-based, inclusive 3:8 maps to Python's 0-based, exclusive 2:8
x01 = Weekly.iloc[:, 2:8]
y01 = (Weekly['Direction'] == 'Up').astype(int)

# Add intercept for statsmodels
X = sm.add_constant(x01)

# Fit logistic regression
glm0_fit = sm.Logit(y01, X).fit(disp=False)
print(glm0_fit.summary())

# Predict probabilities and convert to binary predictions
prob = glm0_fit.predict(X)
pred = (prob > 0.5).astype(int)

# Confusion matrix and accuracy
conf_matrix = pd.crosstab(y01, pred, rownames=['Actual'], colnames=['Predicted'])
print(conf_matrix)

accuracy = (pred == y01).mean()
print(accuracy)

File: task_equivalents/llm_translations/gemini/optimized/r/statistical_modelling_analysis/02_logistic_regression_odds_ratios.R
Weekly <- read.csv('data/csv/Weekly.csv')

x01 <- Weekly[, 3:8]
y01 <- as.integer(Weekly$Direction == 'Up')

glm0_fit <- glm(y01 ~ ., data = cbind(y01, x01), family = binomial)
print(summary(glm0_fit))

prob <- predict(glm0_fit, type = "response")
pred <- as.integer(prob > 0.5)

print(table(y01, pred))
print(mean(pred == y01))